src/conflictresolutionscreen.cpp

```diff
@@ -0,0 +1,138 @@
+    QDBusConnection sysBus = QDBusConnection::systemBus();
+    if ( !sysBus.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    sysBus.connect( QString(), QString(), "org.opensuse.OCIHelper", "displayProblemAndSolutions", this, SLOT( problemSolutionWidget( QString, QString ) ) );
+   
+    setLayout ( m_mainLayout );
+}
+
+void ConflictResolutionScreen::problemSolutionWidget( QString probDescription, QString solutions )
+{
+    m_solId = -1;
+    if ( m_solutionWidget != NULL && m_questionLabel != NULL ) {
+	m_mainLayout->removeWidget( m_questionLabel );
+	m_mainLayout->removeWidget( m_solutionWidget );
+    }
+    QLabel *questionLabel = new QLabel( probDescription );
+    questionLabel->setStyleSheet( "background-color: rgb(254, 250, 210); border-bottom : 1px solid rgb(252,233,79); border-left : 1px solid rgb(196,181,147); border-top : 1px solid rgb(196,181,147); border-right : 1px solid rgb(196,181,147);" );
+    questionLabel->setFixedHeight( 50 );
+    questionLabel->setWordWrap( true );
+    
+    QWidget *solutionWidget = new QWidget;
+    solutionWidget->setObjectName( "solutionWidget" );
+    solutionWidget->setStyleSheet( "QWidget#solutionWidget{ background-color : white; border-bottom: 1px solid rgb(196, 181, 147); border-left : 1px solid rgb(196,181,147); border-top : 1px solid rgb(196,181,147); border-right : 1px solid rgb(196,181,147); }" );
+    
+    QVBoxLayout *solutionLayout = new QVBoxLayout( solutionWidget );
+    solutionLayout->setSpacing( 0 );
+    
+    for ( QString solution : solutions.split( "$", QString::SkipEmptyParts) ) {
+	QRadioButton *sol = new QRadioButton( solution );
+	sol->setStyleSheet( "padding-left: 40px;" );
+	solutionLayout->addWidget( sol );
+	solutionLayout->setSpacing( 10 );
+    }
+    
+    m_questionLabel = questionLabel;
+    m_solutionWidget = solutionWidget;
+    m_mainLayout->addWidget( questionLabel );
+    m_mainLayout->setSpacing( 0 );
+    m_mainLayout->addWidget( solutionWidget );
+}
+
+
+void ConflictResolutionScreen::cancelInstallation() 
+{
+    qDebug() << "cancelling installation";
+    
+    m_ociHelper = new org::opensuse::OCIHelper("org.opensuse.OCIHelper", "/", QDBusConnection::systemBus(), this);
+    if ( !m_ociHelper->isValid() ) {
+	qFatal( "Oops! Cannot connect to the service org.opensuse.OCIHelper" );
+	exit( 1 );
+    }
+    m_ociHelper->killBackend();
+    qApp->quit();
+}
+
+void ConflictResolutionScreen::setSolutionID()
+{
+    /*
+     * Get the main layout  QLayout *layout = QWidget::layout();
+     * Access its children (especially solutionWidget)
+     * Check if QRadioButton is checked - by default it'll be the first one 
+     * Figure something out to uniquely identify the solution (like id number may be)
+     * send it to OCIHelper
+     * 
+     * This is similar to what we've done to disable Install Button when no packages
+     * are selected.
+     */
+    qDebug() << " in sendSolutionToOCIHelper ";
```

Comment:
I think you can replace all that with: m_solId = m_buttonGroup.checkedId(); if you do the previous changes